File: frontend/src/types/telegram.ts
/**
 * Telegram API type definitions.
 */

export interface TelegramCredentials {
  api_id: number;
  api_hash: string;
  phone?: string;
}

export interface Dialog {
  id: number;
  name: string;
  type: string;
  entity_id: number;
  unread_count: number;
}

export interface Sender {
  id: number;
  first_name?: string;
  last_name?: string;
  username?: string;
  phone?: string;
  type: string;
}

export interface Message {
  id: number;
  text: string;
  date: string;
  sender_id?: number;
  sender?: Sender;
  has_media: boolean;
  views?: number;
  forwards?: number;
  reply_to_msg_id?: number;
}

export interface SearchRequest {
  query: string;
  dialog_ids?: number[];
  limit?: number;
}

export interface ApiResponse<T> {
  success: boolean;
  data?: T;
  error?: string;
} 

File: backend/app/schemas/telegram.py
"""
Pydantic schemas for the Telegram API.
"""

from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field

class TelegramCredentials(BaseModel):
    """Telegram API credentials."""
    api_id: int = Field(description="Telegram API ID")
    api_hash: str = Field(description="Telegram API hash")
    phone: Optional[str] = Field(None, description="Phone number for authentication (if needed)")

class DialogResponse(BaseModel):
    """Dialog information model."""
    id: int
    name: str
    type: str
    entity_id: int
    unread_count: int

class MessageResponse(BaseModel):
    """Message information model."""
    id: int
    text: str
    date: str
    sender_id: Optional[int] = None
    sender: Optional[Dict[str, Any]] = None
    has_media: bool
    views: Optional[int] = None
    forwards: Optional[int] = None
    reply_to_msg_id: Optional[int] = None

class SearchRequest(BaseModel):
    """Message search request model."""
    query: str = Field(description="Text to search for")
    dialog_ids: Optional[List[int]] = Field(None, description="List of dialog IDs to search in (None for all)")
    limit: int = Field(100, description="Maximum number of results to return") 